### DISCUSSION TITLE
Include my org or private contributions?

### DISCUSSION THREAD

**Discussion by AlbinoGeek** [lowlighter/metrics#1065] [Q&A]
Hello!

Example: https://github.com/AlbinoGeek/AlbinoGeek

Most of my contributions are in private or organization repositories, but they still show up in my account events. I have the relevent key set in my repository secrets, so I wonder; am I missing anything else?

This seems to effect multiple plugins, hence I figure it's my fault, (heh).

If this falls under https://github.com/lowlighter/metrics/discussions/1056, I apologize for the duplicate.

**Comment by lowlighter** [comment]
Hi!

For the private contributions, it seems to be a reccurent bug so I'll try to investigate when I have some free time 👍 

As for the organization related stuff, maybe you can solve it through this option: [`repositories_affiliations`](https://github.com/lowlighter/metrics/blob/master/source/plugins/base/README.md#repositories_affiliations)
By default it's set to "owner", but if you tweak it with broader affilations you may have better results 🙂

**Reply by AlbinoGeek** [reply]
![image](https://user-images.githubusercontent.com/1910461/170103892-43fc9d90-5e20-467e-a5b0-6f78917fee7d.png)

This seems contrary to the documentation that states it expects an array.

Does "`array` comma separated" actually mean "a comma-separated array as a string"?

That's contrary to say... [`plugin_achievements_ignored`](https://github.com/lowlighter/metrics/blob/master/source/plugins/achievements/README.md#plugin_achievements_ignored) which also describes an `array` and does really take an `array`.

**Reply by AlbinoGeek** [reply]
Nevermind, that's fault of the syntax highlighter here in GitHub. The rendered result works as expected:

```yaml
          repositories_affiliations: >-
            owner, collaborator
```

**Reply by AlbinoGeek** [reply]
After making this change, it only affected some plugins:

## Achievements

https://github.com/AlbinoGeek/AlbinoGeek/commit/bc7a58e66dc3add268498bb21e304fe457d8252b

It found 4 more languages I worked with, cool, okay.

## Habits

Did not work at all. This makes it look like I code for five minutes a week, haha.

## Languages

https://github.com/AlbinoGeek/AlbinoGeek/commit/ba6c00216cfb1dfc514e7e6a21eee231e99917d7

But then I only have 3 more languages here? And none of them rendered.

Same commit count, same code size, same line count.

--

Maybe this is that Public/Private issue?

**Reply by lowlighter** [reply]
Yeah as you guessed when the typing says `array` in the doc, it mostly to indicate that you can pass multiple values (usually comma separated), GitHub Actions only support numbers, strings and yaml 1.2 booleans so typing are parsed by the app itself.

For the languages plugin, might be because you're using the `indepth` mode ?
The achievement one uses the "basic algorithm" of the plugin language (meaning it counts all languages displayed on the sidebar of GitHub repositories, whether you actually write these files or not). The "indepth mode" of the language plugin actually clones your repositories, filter your commits and then run Github linguist on your patches, so it may be normal that it reports "less work" because it should only considers what you've actually worked on. If you disable `indepth` mode I think you should get the same result.

As for habits one I don't know, though it really depends on your activity. When this plugin was written, the assumption was that GitHub actually kept a huge event activity log, but it doesn't seem the case, it's like ~200 events max. So if you comment, make pull requests, create repo, star repo, etc. there will be a lot less "push events" which makes the stat a bit wonky 😅 

Also, each plugin is mostly independant and do not share any information (except the `base` one) so they could report slight differences between them

I'll still look to check if there's some room for improvements though, but I don't really have any solution to offer to you atm 😅 
Hope it provides you more insight though
